The following builds have failed:
- [Pull Request Feedback (Trigger) (Check)](https://builds.gradle.org/buildConfiguration/Gradle_Master_Check_Stage_PullRequestFeedback_Trigger/115962168)
  - [Build Scan](https://ge.gradle.org/scans?search.buildOutcome=failure&amp;search.tags=CI&amp;search.names=gitCommitId&amp;search.values=8d443dc3dd58063d1fd0f79c24b1de7b61daad27&amp;)

<details><summary>🔎 3 failure groups (1 build, 2 test)</summary>

🧪 **`JavaConfigurationPerformanceTest.configure`** in `:performance:largeJavaMultiProjectKotlinDslPerformanceTest` · 1 occurrence · [scan](https://ge.gradle.org/s/6hup2pg3acceu/tests/overview?outcome=failed)
> Condition failed with Exception:
> 
> result.assertCurrentVersionHasNotRegressed()
> |      |
> |      java.lang.AssertionError: Speed Results for test project 'largeJavaMultiProjectKotlinDsl' with tasks help: we're slower than 9.8.0-commit-41203b0be36 with 99% confidence.
> |      Difference: 96 ms slower (96 ms), 4.58%
> |        Current Gradle median: 2.19 s min: 2.066 s, max: 2.556 s, se: 135.93 ms}
> |        > [2.532 s, 2.502 s, 2.369 s, 2.498 s, 2.445 s, 2.556 s, 2.35 s, 2.405 s, 2.22 s, 2.399 s, 2.216…
🛠️ **`Failed to stop service 'testFilesCleanupBuildService'.`** · 1 occurrence · [scan](https://ge.gradle.org/s/ro63tzqspkmzc/failures)
> Found non-empty test files dir:
> /home/tcagent1/agent/work/f63322e10dd6b396/platforms/software/version-control/build/tmp/teŝt files/ParallelSou.Test:
>  n2v40/.gradle/configuration-cache/gc.properties
>  n2v40/.gradle/9.8.0-20260805153655+0000/gc.properties
>  n2v40/.gradle/9.8.0-20260805153655+0000/fileChanges/last-build.bin
🧪 **`(N/A).Partition 62 in session 4 on gradle-enterprise-test-distribution-agent-b6fb8f869-m7jzp`** in `:dependency-management:isolatedProjectsIntegTest` · 1 occurrence · [scan](https://ge.gradle.org/s/r2d5od4ayeu3i/tests/overview?outcome=failed)
> java.lang.RuntimeException: Failure during test execution: Forked test JVM terminated unexpectedly with exit value 137
> For potential reasons why the test JVM exited, check the troubleshooting section at https://gradle.com/help/test-distribution-troubleshooting.
</details>

<sub>Was this summary helpful? [Good 👍](https://gradle-bot.grdev.net/feedback?pr=38675&build=115962168&rating=positive) · [Bad 👎](https://gradle-bot.grdev.net/feedback?pr=38675&build=115962168&rating=negative) · or [report an issue in #bt-dev-prod](https://gradle.slack.com/archives/CBSJ2GUTV)</sub>

--- a/platforms/documentation/docs/src/docs/userguide/optimizing-builds/configuration-cache/configuration_cache_requirements.adoc
+++ b/platforms/documentation/docs/src/docs/userguide/optimizing-builds/configuration-cache/configuration_cache_requirements.adoc
@@ -722,9 +722,9 @@ When process isolation is used, the worker action is executed in a separate work
 
 In simple cases, when the libraries also provide command-line entry points (`public static void main()` method), you can also use the link:{javadocPath}/org/gradle/api/tasks/JavaExec.html[`JavaExec`] task to isolate the library.
 
-The agent also affects hot-swapping build logic classes in a running build.
-It prevents Hot Code Replace of such classes: Gradle logs a warning and ignores the recompiled definition.
-You should not rely on Hot Code Replace when debugging build logic; restart the build to pick up such a change.
+Gradle's instrumentation agent also affects hot-swapping build logic classes in a running build.
+Gradle serves pre-instrumented bytecode for buildscript and plugin classes, so a Hot Code Replace of such a class is ignored: Gradle logs a warning and keeps the pre-instrumented definition.
+You should not rely on Hot Code Replace when debugging build logic; restart the build to pick up your edit.
 
 [[config_cache:secrets]]
 == Handling of Credentials and Secrets
@@ -791,6 +791,9 @@ $ export GRADLE_ENCRYPTION_KEY="your-generated-key-here"
 The Configuration Cache supports third-party `-javaagent:` attachments to the build JVM (for example a coverage agent such as Jacoco, IntelliJ's `idea_rt.jar`, or a custom user agent) in regular daemon builds and in <<test_kit#test_kit, TestKit>>'s default (daemon) mode.
 
 Only agents attached at JVM startup are supported. Agents attached dynamically via the Attach API after the JVM has started are not detected and may cause incorrect bytecode observation on classes loaded after attachment.
+
+The JDWP debug agent and async-profiler are recognized and do not trigger third-party agent handling, so a debugging or profiling session runs the same instrumentation path as a regular build.
+Note that debugger-initiated Hot Code Replace of instrumented buildscript or plugin classes is not honored, see <<config_cache:requirements:bytecode_modifications_and_isolation, Bytecode Modifications and Java Agent>> for details.
 
 It is not supported when TestKit runs in embedded mode (`withDebug(true)`). To use a third-party Java agent with the Configuration Cache from there, drop `withDebug(true)` and let TestKit use its daemon default.
 

